src/core/server/http/router/router.ts

```diff
@@ -273,6 +288,21 @@ export class Router implements IRouter {
   }
 }
 
+const convertEsUnauthorized = (e: EsNotAuthorizedError): ErrorHttpResponseOptions => {
+  return {
+    body: e.message,
+    headers: Object.entries(e.headers).reduce((headers, [key, value]) => {
+      if (key.toLowerCase() === 'www-authenticate') {
+        return {
+          ...headers,
+          'www-authenticate': value,
```

Comment:
the legacy code has fallen back to `'Basic realm="Authorization Required"'` https://github.com/elastic/kibana/blob/6164037f18b4d715f16bc42fc932576d089692c3/src/core/server/elasticsearch/legacy/errors.ts#L87